tee-worker/litentry/core/assertion-build/src/holding_time.rs

```diff
@@ -239,9 +240,9 @@ fn search_holding_date(
 ) -> core::result::Result<Option<&'static str>, ErrorDetail> {
 	let mut client = AchainableClient::new(data_provider_config);
 
-	let mut pred = |date| {
+	let mut pred = |date: &&str| {
 		let (outcome, new_accounts) =
-			holding_time_search_step(&mut client, q_min_balance, accounts, date);
+			holding_time_search_step(&mut client, q_min_balance, accounts.clone(), *date);
 		accounts = new_accounts;
 		outcome.map(|is_holding| !is_holding) // negated to match the partition_point API
 	};
```

Comment:
Thanks @Kailai-Wang for fixing this issue! Still gotta wrap my head around some of the subtleties here, apparently :D

In this case the extra `clone` shouldn't make a huge difference overall, since the cost is in any case dominated by the HTTP requests; in general though it does feel a bit wasteful to have to clone the array each time, only to replace the original anyway after the call to `holding_time_search_step`. What would (in theory) have been the better approach here? (Not for this PR to be sure, just as an overall learning)